### PR TITLE
[Shader Management] Setting up Test Project for Foundry with Inline Tests (OLD)

### PR DESCRIPTION
**Pull request by cdxntchou**
### Purpose of this PR
Adding ShaderFoundry integration tests in Graphics Repo -- with inline graphics testing framework

---
### Testing status

Added 4 tests that verify basic functionality of the test framework, success and failure conditions are detected.
Ran tests locally (DX11 / PC) -- passed, verified tests are executing.

Running yamato job:

DX11: 🟢 - verified execution
https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/sg2%252Fshader-sandbox-ci-tests/.yamato%252Fshaderfoundry-win-dx11.yml%2523ShaderFoundry_Win_DX11_editmode_mono_Linear_CUSTOM-REVISION/11243329/job/pipeline

Full PR ShaderFoundry:
https://yamato.cds.internal.unity3d.com/jobs/902-Graphics/tree/sg2%252Fshader-sandbox-ci-tests/.yamato%252Fall-shaderfoundry.yml%2523PR_ShaderFoundry_CUSTOM-REVISION/11327110/job/pipeline

Extra ShaderFoundry:

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

**Review by joshua-davis — APPROVED** [review]

File: TestProjects/ShaderFoundry/Assets/CommonAssets/Editor/FoundryGraphicsTestsEditor.asmdef [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+    "name": "Unity.ShaderGraph.Editor.GraphicsTests",
```
Comment (joshua-davis) [inline TestProjects/ShaderFoundry/Assets/CommonAssets/Editor/FoundryGraphicsTestsEditor.asmdef:2]: This assembly name should probably be `Unity.ShaderFoundry.Editor.GraphicsTests`

Comment (cdxntchou) [reply TestProjects/ShaderFoundry/Assets/CommonAssets/Editor/FoundryGraphicsTestsEditor.asmdef:2]: Yeah, I had it changed to ShaderFoundry, but we use one ShaderGraph internal class, that I didn't feel like copying out (PreviewResources) -- so it's currently being granted internal access via the name...  we could either copy that class out, or use asmref hackery to get access to it...

File: TestProjects/ShaderFoundry/Assets/CommonAssets/Editor/FoundryGraphicsTestsEditor.asmdef [hunk 2]
```diff
@@ -1,9 +1,8 @@
 {
-    "name": "Unity.ShaderGraph.Editor.GraphicsTests",
+    "name": "Unity.Foundry.Editor.GraphicsTests",
```
Comment (joshua-davis) [inline TestProjects/ShaderFoundry/Assets/CommonAssets/Editor/FoundryGraphicsTestsEditor.asmdef:2]: nit: "ShaderFoundry" instead of "Foundry"?

**Comment by cdxntchou** [comment]
Need to rebase on top of shader-foundry/staging

**Comment by cdxntchou** [comment]
Rebased this off of shader-foundry/staging.   New PR here:  https://github.com/Unity-Technologies/Graphics/pull/6992

**Pull request closed without merging**